test/transcoding/SPV_INTEL_hw_thread_queries/intel_hw_thread_queries.ll

```diff
@@ -0,0 +1,39 @@
+; Test for SPV_INTEL_hw_thread_queries feature
+; SPIR-V friendly LLVM IR to SPIR-V translation and vice versa
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_hw_thread_queries -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+; RUN: llvm-spirv -r --spirv-target-env=SPV-IR %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: Capability HWThreadQueryINTEL
+; CHECK-SPIRV: Extension "SPV_INTEL_hw_thread_queries"
+; CHECK-SPIRV: Decorate [[#Id1:]] BuiltIn 6135
+; CHECK-SPIRV: Decorate [[#Id2:]] BuiltIn 6136
+; CHECK-SPIRV: Decorate [[#Id3:]] BuiltIn 6137
+; CHECK-SPIRV: Variable [[#]] [[#Id1]]
+; CHECK-SPIRV: Variable [[#]] [[#Id2]]
+; CHECK-SPIRV: Variable [[#]] [[#Id3]]
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+@__spirv_BuiltInSubDeviceIDINTEL = external addrspace(1) global i32
+@__spirv_BuiltInHWThreadIDINTEL = external addrspace(1) global i32
+@__spirv_BuiltInMaxHWThreadIDPerSubDeviceINTEL = external addrspace(1) global i32
+
+; Function Attrs: nounwind readnone
+define spir_kernel void @foo() {
+entry:
+  %0 = load i32, i32 addrspace(4)* addrspacecast (i32 addrspace(1)* @__spirv_BuiltInSubDeviceIDINTEL to i32 addrspace(4)*), align 4
+  %1 = load i32, i32 addrspace(4)* addrspacecast (i32 addrspace(1)* @__spirv_BuiltInHWThreadIDINTEL to i32 addrspace(4)*), align 4
+  %2 = load i32, i32 addrspace(4)* addrspacecast (i32 addrspace(1)* @__spirv_BuiltInMaxHWThreadIDPerSubDeviceINTEL to i32 addrspace(4)*), align 4
+  ; CHECK-LLVM: call spir_func i32 @_Z31__spirv_BuiltInSubDeviceIDINTELv() #1
```

Comment:
Can we also check that LLVM IR with calls but not builtin variables is correctly recognized by the translator?

Reply:
Added a separate test for that. But it does look like if in the module there are both representations of builtins simultaneously - it confuses the translator and it generated not correct SPIR-V (not only for this extension). Please consider following example:
```
@__spirv_BuiltInSubgroupMaxSize = external addrspace(1) global i32
define spir_kernel void @foo() {
entry:
  %0 = call spir_func i32 @_Z30__spirv_BuiltInSubgroupMaxSizev() #1
  ret void
}
define spir_kernel void @boo() {
entry:
  %0 = load i32, i32 addrspace(4)* addrspacecast (i32 addrspace(1)* @__spirv_BuiltInSubgroupMaxSize to i32 addrspace(4)*)
  ret void
}
declare dso_local spir_func i32 @_Z30__spirv_BuiltInSubgroupMaxSizev()
```
 Translates to:
```
 12 Decorate 4 LinkageAttributes "__spirv_BuiltInSubgroupMaxSize" Import
 4 Decorate 4 BuiltIn 37
 13 Decorate 6 LinkageAttributes "__spirv_BuiltInSubgroupMaxSize.1" Import
 3 Decorate 6 Constant
```
and then to:
```
%0 = load i32, i32 addrspace(7)* @__spirv_BuiltInSubgroupMaxSize.1, align 4
// ...
%0 = call spir_func i32 @_Z22get_max_sub_group_sizev()
```
I believe, it might relates to https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/978 .